### PR TITLE
Reorganize Protobuf package structure

### PR DESCRIPTION
**Pull request by christophermaier**
### What changes does this PR make to Grapl? Why?

Imposes a more standard and future-proof organizational scheme to our Protobuf definitions.

The current Protobuf definitions were moved from the `graph_definitions` package to the new `graplinc.grapl.api.graph.v1beta1` package.

This gives us a more natural hierarchy in which to locate our types, and also provides a way to deal with backwards incompatibilities we may choose to introduce in the future.

Rust and Python code were updated to deal with the new packaging structure. In particular, the Python distribution for this generated code has been reorganized into a [namespace package](https://packaging.python.org/guides/packaging-namespace-packages/). The module structure mirrors that of the protobufs, but the `graplinc.grapl` portion only provides a _namespace_; the actual distributed code, now called `graplinc-grapl-api` will be the package that deals with only the Protobufs related to the Grapl API. Future Protobufs for other parts of Grapl (or even other products) would be distributed in their own packages. Thanks to namespace packages, each of these distributions can be delivered independently and will not conflict with each other. Incidentially, there is no reason that we would need to limit this to only Protobuf-related code; _all_ our Python code could eventually be distributed in this mechanism (e.g., `grapl-common` could become `graplinc-grapl-common`, etc.)

To make this more explicit, this PR _removes_ the `grapl-graph-descriptions` Python package in favor of a new `graplinc-grapl-api` package. This is a breaking change (as is the Protobuf package reorganization itself), but the sooner done, the better.

The Rust types could probably be reorganized to better map onto the new Protobuf structure, but that was not dealt with here.

In addition to restructuring the Protobuf package structure, this PR also moves the Protobuf definitions out of the `src/rust/graph-definitions` project and up to a new `src/proto` directory. The Python code for generating the aforementioned `graplinc-grapl-api` package has also been moved from `src/rust/graph-definitions` over to the `src/python` directory where it belongs. This structure is much less "surprising" than the previous layout, and will help to better reason about the code.

Finally, this is a good time to introduce linting around our Protobufs. Here, we use [buf](https://buf.build) to perform both linting and backwards-compatibility checking on our Protobufs, which will help bring rigor and overall stability to our definitions as they evolve.

(Discussed internally at https://grapl-internal.slack.com/archives/C0180LW59ED/p1612463795214900, etc.)

### How were these changes tested?

`make test`, and lots of manual testing.

### CODE REVIEW

**Comment by christophermaier** [comment]
Note to reviewers: both of the newly-added protobuf lints are expected to fail at this point. The `lint` check is flagging a field name that needs to be changed, and the `breaking` check is comparing against what's currently on `staging`, which we don't really care about right now since this is a big change anyway. Once this is merged, and we address that field name, we can make both these checks required for all future merges.

**Review by wimax-grapl — APPROVED** [review]

File: .github/workflows/grapl-lint.yml [hunk 1]
```diff
@@ -8,6 +8,31 @@ on:
       - master
 
 jobs:
+  proto-lint:
```
[the remark this answers is not in the document]
Comment (christophermaier) [reply .github/workflows/grapl-lint.yml:11]: @wimax-grapl I can allow this exception for now... are you aware of a specific reason that we have a single field with a camelCase name, though?

Ultimately, I'd like us to be consistently snake_case, but didn't want to change things here (I _think_ this field may be referenced in the web UI?)

File: src/python/graplinc-grapl-api/setup.py [hunk 2]
```diff
@@ -68,24 +83,27 @@ def compile_protobuf(proto_file, proto_dir, out_dir):
...
 def find_proto_files():
-    """find all the .proto files"""
-    for (base_path, _, files) in os.walk(PROTO_DIR):
+    """Find all the .proto files under our protobuf hierarchy"""
+    for (base_path, _, files) in os.walk(OUR_PROTO_PATH):
         for file_name in files:
             if file_name.endswith(".proto"):
-                yield os.path.join(base_path, file_name)
+                full_path = os.path.join(base_path, file_name)
+                print(f"Found protobuf file {full_path}")
```
Comment (jgrillo-grapl) [inline src/python/graplinc-grapl-api/setup.py:91]: 👍

**Review by jgrillo-grapl — APPROVED** [review]
My apologies for that gnarly setup.py, thanks for cleaning it up!

LGTM

**Pull request merged**